### ISSUE TITLE
FR: Support Quod Libet

### ISSUE THREAD

**Issue by mjpieters** [beardedspice/beardedspice#971]
[Quod Libet](https://quodlibet.readthedocs.io/en/latest/) is a cross-platform audio player, and I _really_ miss being able to to control playback with media keys since upgrading to MacOS Catalina.

Could BeardedSpice add support for this?

You can trivially control the app [via a FIFO pipe](https://quodlibet.readthedocs.io/en/latest/guide/interacting.html#controlling-the-player), just write out [one of the supported messages](https://github.com/quodlibet/quodlibet/blob/b92e08484e44f7e056a3fb3f7daeb70b201698c9/quodlibet/cli.py#L76-L82) (encoded to UTF-8 bytes) to the pipe. For the functions that BeardedSpice supports, only the `next`, `previous`, and `play-pause` commands would be needed. A command ends with a newline (don't forget to flush the buffer if you opened the FIFO with anything other than unbuffered or a line buffer).

You can also read the current playing track, from `~/.quodlibet/current`, which contains the full set of [track tags](https://quodlibet.readthedocs.io/en/latest/guide/tags/tags.html), newline-separated, encoded to UTF-8. It should be trivial to extract the `title` and `artist` fields; each line starts with `fieldname=` (where field names _can_ start with `~` or `~#` but otherwise consist of lower-case ascii letters and underscores).

Basically:

- Quod Libet is active if `~/.quodlibet/control` exists and is a FIFO pipe (`S_ISFIFO(path)` is true).
- You can control playback by writing UTF-8 encoded commands to the FIFO pipe, terminated by a newline.
- You can get data on the currently playing song by reading `~/.quodlibet/current`.

My Objective-C fu is practically non-existing, so I am not sure if I could create a competent pull request for this myself.

**Comment by vityevato** [comment]
@mjpieters 
I released a some differently release of the Beardie, by the way :)
https://github.com/Stillness-2/beardie

**Comment by mjpieters** [comment]
> @mjpieters
> I released a some differently release of the Beardie, by the way :)
> https://github.com/Stillness-2/beardie

Thanks! I've since switched to Beardie; I added it to my [personal homebrew tap recently](https://github.com/mjpieters/homebrew-tap/blob/master/Casks/beardie.rb). Unfortunately, Beardie has not yet reached enough watchers to qualify for inclusion in the main Homebrew Casks repository!
